src/lib/libast/comp/iconv.c

```diff
@@ -523,7 +523,7 @@ if (error_info.trace < DEBUG_TRACE) sfprintf(sfstderr, "%s: debug%d: AHA#%d _ast
 		else if (!(cp = (const _ast_iconv_list_t*)ccmaplist((_ast_iconv_list_t*)cp)))
 			cp = codes;
 	}
-	if (cp = bp)
+	if (cp == bp)
```

Comment:
This is not a typo. It's a pattern used in ksh code for assignment.

Reply:
Pattern is one thing.
I need more than that.

Both `cp` and `bp` get assigned up to that line (line 526).

`bp` is `0` at the beginning, line 489.
`cp` starts out as a list, so it points to something, line 491.

`bp` may be assigned to the current value of `cp` in lines 509 and 514; it may still be `0`.
`cp` may be assigned a new value in line 524.

So, if `bp` remained `0` - never assigned anything, it would overwrite `cp` in line 526.
`bp` is not used past line 526.

In fact, this is equivalent code
```c
if (bp != 0) {
 // use bp instead of cp
 //....
}
```
Why assign it to `cp` if you can go ahead with `bp`.

Reply:
@siteshwar is correct although I think of it as an anti-pattern. As I restyle each module If add additional parens to silence the compiler warning and make it clear that it should be an assignment rather than equality test. Or, in a case like this, I'll generally hoist the assignment out of the `if`; e.g.,:

```
cp = bp;
if (cp) {
```